### PR TITLE
Bump rdoc to 6.0

### LINKED ISSUE #6599: Update dependency constraint to allow for rdoc v6.0.0
Hey there! :wave:

I noticed that the constraint you have for rdoc doesn't allow for the latest version to be used.

The constraint I found was `~> 5.0`, and the latest version available is `6.0.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by jekyllbot**
Fix #6599

### CODE REVIEW

**Comment by DirtyF** [comment]
image

**Comment by DirtyF** [comment]
Note:  `rdoc` is an optional dependency and now requires Ruby >= 2.2.2

File: Gemfile [hunk 1]
```diff
@@ -75,7 +75,7 @@ group :jekyll_optional_dependencies do
   gem "jekyll-redirect-from"
   gem "kramdown", "~> 1.14"
   gem "mime-types", "~> 3.0"
-  gem "rdoc", "~> 5.0"
+  gem "rdoc", "~> 6.0"
```
Comment (pathawks) [inline Gemfile]: Maybe still allow `5.x` for Ruby 2.1?

```
gem "rdoc", [">= 5.0", "< 7.0"]

Comment (ashmaroli) [reply Gemfile]: like we have for other gems in similar situations.
```ruby
gem "rdoc", RUBY_VERSION >= "2.2.2" ? "~> 6.0" : "~> 5.1"
```

**Comment by parkr** [comment]
We've dropped Ruby 2.1, if that changes anything.

**Comment by parkr** [comment]
@jekyllbot: merge +dev

**Pull request merged**